src/Bridges/Constraint/bridges/ExponentialConeToScalarNonlinearFunctionBridge.jl

```diff
@@ -0,0 +1,159 @@
+# Copyright (c) 2017: Miles Lubin and contributors
+# Copyright (c) 2017: Google Inc.
+#
+# Use of this source code is governed by an MIT-style license that can be found
+# in the LICENSE.md file or at https://opensource.org/licenses/MIT.
+
+"""
+    ExponentialConeToScalarNonlinearFunctionBridge{T,F} <:
+        Bridges.Constraint.AbstractBridge
+
+`ExponentialConeToScalarNonlinearFunctionBridge` implements the following
+reformulation:
+
+  * ``(x, y, z) \\in \\textsf{ExponentialCone}()`` to
+    ``y \\cdot exp(x / y)) - z \\le 0``, ``y \\ge 0``.
+
+## Source node
+
+`ExponentialConeToScalarNonlinearFunctionBridge` supports:
+
+  * `F` in [`MOI.ExponentialCone`](@ref)
+
+## Target nodes
+
+`ExponentialConeToScalarNonlinearFunctionBridge` creates:
+
+  * [`MOI.ScalarNonlinearFunction`](@ref) in [`MOI.LessThan{T}`](@ref)
+  * [`MOI.ScalarAffineFunction`](@ref) in [`MOI.GreaterThan{T}`](@ref)
+"""
+mutable struct ExponentialConeToScalarNonlinearFunctionBridge{
+    T,
+    F<:Union{MOI.VectorOfVariables,MOI.VectorAffineFunction{T}},
+} <: AbstractBridge
+    f::F
+    ci::MOI.ConstraintIndex{MOI.ScalarNonlinearFunction,MOI.LessThan{T}}
+    ci_y::MOI.ConstraintIndex{MOI.ScalarAffineFunction{T},MOI.GreaterThan{T}}
+end
+
+const ExponentialToScalarNonlinearFunction{T,OT<:MOI.ModelLike} =
+    SingleBridgeOptimizer{ExponentialConeToScalarNonlinearFunctionBridge{T},OT}
+
+function bridge_constraint(
+    ::Type{ExponentialConeToScalarNonlinearFunctionBridge{T,F}},
+    model::MOI.ModelLike,
+    f::F,
+    s::MOI.ExponentialCone,
+) where {T,F<:Union{MOI.VectorOfVariables,MOI.VectorAffineFunction{T}}}
+    x, y, z = MOI.Utilities.scalarize(f)
+    g_x_div_y = MOI.ScalarNonlinearFunction(:/, Any[x, y])
+    g_exp_x_div_y = MOI.ScalarNonlinearFunction(:exp, Any[g_x_div_y])
+    g = MOI.ScalarNonlinearFunction(
+        :-,
+        Any[MOI.ScalarNonlinearFunction(:*, Any[y, g_exp_x_div_y]), z],
+    )
+    ci = MOI.add_constraint(model, g, MOI.LessThan(zero(T)))
+    ci_y = MOI.add_constraint(model, one(T) * y, MOI.GreaterThan(zero(T)))
```

Comment:
Can you add a comment to remind why we don't add a variable bound here ?

Reply:
I remember that it gets very complicated in case it already has a bound if we implement deletion, we need to do it in `final_touch` and so on. We already discussed this in another bridge but I don't remember which one and what we ended up doing. Ideally we should have a small note in the dev doc of MOI that is referenced from every bridge that does that.